### ISSUE TITLE
The max_subs reload unit test is incorrect

### ISSUE THREAD

**Issue by ColinSullivan1** [nats-io/nats-server#4173]
## Defect (Minor)

The server actually does support changing `max_subs` in a reload and will boot any connections in violation.

The [unit test](https://github.com/nats-io/nats-server/blob/5db57fb053a03dd84928ed84fb14da9e2267d1eb/server/reload_test.go#L1752) should be fixed. The test is actually failing on the change of port (set to -1 - randomly generated), not the change in max_subs as was assumed.  Other tests may have this issues as well.

Here's server output of a reload reducing the max connections in an account. A connection now in violation is closed.
```text
[48467] 2023/05/16 15:31:10.539316 [DBG] Trapped "hangup" signal
[48467] 2023/05/16 15:31:10.539536 [INF] Reloaded: authorization users
[48467] 2023/05/16 15:31:10.539543 [INF] Reloaded: accounts
[48467] 2023/05/16 15:31:10.539582 [INF] Reloaded server configuration
[48467] 2023/05/16 15:31:10.539601 [ERR] 127.0.0.1:60414 - cid:6 - maximum subscriptions exceeded
[48467] 2023/05/16 15:31:10.539795 [DBG] 127.0.0.1:60414 - cid:6 - Client connection closed: Client Closed
```

I can try to get to fixing this when I have time, but am certainly OK with the case someone beats me to it.

#### Versions of `nats-server` and affected client libraries used:
nats-server: v2.9.17-beta.5

**Comment by derekcollison** [comment]
I think we just remove this test which is testing that its not supported no?

**Comment by ColinSullivan1** [comment]
True, could remove, but was thinking of changing the test such that reload with max_subs changes behaves as expected.
